NDTensors/src/blocksparse/blocksparse.jl

```diff
@@ -12,21 +14,29 @@ struct BlockSparse{ElT,VecT,N} <: TensorStorage{ElT}
   end
 end
 
-# TODO: Implement as `fieldtype(storagetype, :data)`.
-datatype(::Type{<:BlockSparse{<:Any,DataT}}) where {DataT} = DataT
-# TODO: Implement as `ndims(blockoffsetstype(storagetype))`.
-ndims(storagetype::Type{<:BlockSparse{<:Any,<:Any,N}}) where {N} = N
 # TODO: Implement as `fieldtype(storagetype, :blockoffsets)`.
 blockoffsetstype(storagetype::Type{<:BlockSparse}) = BlockOffsets{ndims(storagetype)}
 
-function set_datatype(storagetype::Type{<:BlockSparse}, datatype::Type{<:AbstractVector})
-  return BlockSparse{eltype(datatype),datatype,ndims(storagetype)}
+function TypeParameterAccessors.position(
+  ::Type{<:BlockSparse}, ::typeof(TypeParameterAccessors.parenttype)
+)
+  return Position(2)
+end
+TypeParameterAccessors.position(::Type{<:BlockSparse}, ::typeof(Base.ndims)) = Position(3)
```

Comment:
```suggestion
TypeParameterAccessors.position(::Type{<:BlockSparse}, ::typeof(ndims)) = Position(3)
```

Reply:
This was failing because we have a version of `NDTensors.ndims` I fixed (for now) the NDTensors version so I reverted back to `::typeof(ndims)`

Reply:
Thanks, I had forgotten that we had our own `NDTensors.ndims` function, I can see how that complicates these definitions a bit and I didn't realize you were using `Base.ndims` to distinguish from `NDTensors.ndims`. I made a suggestion in another comment to move `NDTensors.ndims` to `TypeParameterAccessors`, which could help keep those two versions straight, since then we can use `TypeParameterAccessors.ndims` to make it clear we don't mean to use `Base.ndims`.